### DISCUSSION TITLE
nbb.edn

### DISCUSSION THREAD

**Discussion by jaidetree** [babashka/nbb#134] [Ideas]
Curious what an nbb.edn would look like and do.

Have a few ideas off the top of my head:

- Define src paths to load into the classpath like bb.edn's paths attr
- Define cljs deps that nbb installs without having to use Java. 
- Perhaps some aliases for different configs? Though may be redundant to package.json scripts.

As for installing deps, I think you were considering leveraging git repos. That should work as I know both npm and yarn support direct github repo URLs.

Alternatively might be able to port some of deps.edn logic for nbb if it's worth owning ourselves.

**Comment by avelino** [comment]
I like the idea of having `nbb.edn` as it is `shadow-cljs.edn`, but it bothers me having to maintain two files `nbb.edn` and `package.json` (from node), we can think one step further and make `nbb.edn` be able to replace `package.json`

cc @borkdude

**Reply by borkdude** [reply]
Or the other way around: have some nbb settings inside package.json?

**Reply by avelino** [reply]
I like the idea, to have node specifying nbb configuration
I believe this solution simplifies the implementation and we don't get away from the [Node specification](https://docs.npmjs.com/cli/v8/configuring-npm/package-json)

**Reply by borkdude** [reply]
On the other hand, I don't really like editing JSON, so I think having an `nbb.edn` separately from `package.json` would have benefits too. It would also have the benefit of not having to parse `package.json` on startup to look for settings, but just checking if the `nbb.edn` file exists, which is cheaper.

**Reply by avelino** [reply]
I like to work with small steps (baby steps)

I believe that reading the `nbb.edn` configuration nodes into package.json is less ~"costly"~ than thinking up a specification for `nbb.edn` (so as not to depend on `package.json`)

we can start with using `package.json` and then implement `nbb.edn`

**Reply by borkdude** [reply]
I don't follow. The nodes in the package.json or in a nbb.edn file both are a specification.
